engine/util/mode.go

```diff
@@ -0,0 +1,297 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type ModeError string
+
+func (m ModeError) Error() string { return string("invalid mode: " + m) }
```

Comment:
https://github.com/purpleidea/mgmt/blob/master/docs/style-guide.md#method-receiver-naming please.

Reply:
Note: I'm not actually sure why you're defining a special error type unless you want to check against it somewhere. Eg:

https://github.com/purpleidea/mgmt/blob/06952c224b734b5146539d1fedc953e7b1a50041/lang/interfaces/error.go#L29

If you're planning to, go ahead, if not you can just use: `fmt.Errorf` in the error returns.